flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1101,14 +1101,20 @@ public CompletableFuture<Acknowledge> shutDownCluster(
     @Override
     public CompletableFuture<CoordinationResponse> deliverCoordinationRequestToCoordinator(
             JobID jobId,
-            OperatorID operatorId,
+            String operatorUid,
             SerializedValue<CoordinationRequest> serializedRequest,
             Duration timeout) {
+        // Convert operatorUid to OperatorID for querying.
+        // This is possible because once an operator is assigned an operatorUid,
+        // that UID takes precedence during the generation of the JobGraph
+        // and is directly used to generate the OperatorID.
```

Comment:
> This is possible ...

This approach is feasible because an operator's OperatorID is derived from its UID, when available.